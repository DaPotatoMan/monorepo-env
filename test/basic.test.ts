import { fileURLToPath } from 'node:url'
import { describe, expect, it } from 'vitest'
import { $fetch, setup } from '@nuxt/test-utils/e2e'

describe('ssr', async () => {
  await setup({
    rootDir: fileURLToPath(new URL('../playground', import.meta.url)),
  })

  it('renders the index page', async () => {
    const html = await $fetch('/api/get-env')
    expect(html).toContain('VITE_APP_MODE')
    expect(html).toContain('VITE_APP_IS_LOCAL')

    const runtimeEnv = await $fetch('/api/get-runtime-env')
    expect(runtimeEnv).toMatchSnapshot()
  })
})
