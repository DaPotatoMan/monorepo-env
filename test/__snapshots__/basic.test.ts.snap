// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ssr > renders the index page 1`] = `
{
  "appMode": "prod",
  "key": "my-private-key",
  "public": {
    "key": "my-public-key",
  },
}
`;
